src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp

```diff
@@ -610,6 +618,82 @@ class TDatabaseFeedManager final : public Observer<nlohmann::json&>
         }
     }
 
+    /**
+     * @brief Checks and translates package information.
+     *
+     * This function searches for translation data in the Level 1 and Level 2 caches for a given package.
+     * If translation data is found, it populates the translations and returns them. If no translation data
+     * is found, it logs a debug message and returns an empty vector.
+     *
+     * @param package The package data to be checked and translated.
+     * @param osPlatform The operating system platform for which the package translation is required.
+     * @return A vector containing the translated package data.
+     */
+    std::vector<PackageData> checkAndTranslatePackage(const PackageData& package, const std::string& osPlatform)
+    {
+        std::vector<PackageData> vulnerabilityTranslations;
+        const auto cacheKey = osPlatform + "_" + package.vendor + "_" + package.name;
+
+        auto translatePackage = [&](const auto& translations)
+        {
+            for (const auto& translation : translations)
+            {
+                PackageData translatedPackage = package;
+                if (!translation.translatedProduct.empty())
+                {
+                    translatedPackage.name = translation.translatedProduct;
+                }
+                if (!translation.translatedVendor.empty())
+                {
+                    translatedPackage.vendor = translation.translatedVendor;
+                }
+                if (!translation.translatedVersion.empty())
+                {
+                    translatedPackage.version = translation.translatedVersion;
+                }
+                vulnerabilityTranslations.push_back(translatedPackage);
+            }
+        };
+
+        // Check Level 1 cache
+        if (m_translationFilter->count(cacheKey) == 0 && m_translationL1Cache->isHit(cacheKey))
+        {
+            logDebug2(WM_VULNSCAN_LOGTAG,
+                      "Translation for package '%s' on platform '%s' found in Level 1 cache.",
+                      package.name.c_str(),
+                      osPlatform.c_str());
+
+            const auto& L1Translations = m_translationL1Cache->getValue(cacheKey).value();
+            translatePackage(L1Translations);
+            return vulnerabilityTranslations;
+        }
+
+        // Check Level 2 cache
+        const auto L2Translations = getTranslationFromL2(package, osPlatform);
+        if (!L2Translations.empty())
```

Comment:
What if the translation is in the filter?

Reply:
According to the design solution proposal, we form a pipeline to search for the translation first in the L1, then L2 and if no translation was found, add it to the filter 😃 

```mermaid
---
title: Translation flow
---
stateDiagram-v2
    [*] --> SearchL1
    SearchL1 --> SearchL2
    SearchL2 --> Filter
    SearchL2 --> [*]
    Filter --> [*]
    SearchL1 --> [*]
```

Reply:
Yes but if the translation is already filtered, we are looking in L2 anyways

Reply:
Yes, but at the same time, we are losing readability (we could add a specific log for the first check)